### ISSUE TITLE
My attempt to repair Aleona's Tales dragon sprites

### ISSUE THREAD

**Issue by SenhorFlibble** [Andrettin/Wyrmsun#195]
Apologies once again for possibly using the issues system for a wrong purpose, but I know no other way to communicate with you.

I've attempted to improve the dragon sprites from Aleona's Tales using the ESRGAN neural network, specifically using a model called Pixel Perfect (4x_PixelPerfectV4_137000_G.pth). It upscales images to 4x the original size and smoothens the dithering. I then downsized and palettised the result: [original colours](https://imgur.com/ep8dDz8), [team colours](https://imgur.com/QCGDW6G).

Since Wyrmsun does not appear to have a flying dragon unit yet, perhaps you could make use of it?

**Comment by Andrettin** [comment]
Please, there's nothing to apologize for, it's perfectly fine to use issues for these, well, issues :)

Thanks for sharing the dragon! I think it wouldn't fit in very well with the game's existing graphics, though. But it would be a cool resource for modders.

**Comment by SenhorFlibble** [comment]
Speaking of modder resources, I've been digging the Stratagus forums (sadly abandoned, as it seems), and discovered Nikita Sadkov's posts from 2016 where he, among other things, [discovered](https://forums.stratagus.com/viewtopic.php?f=33&t=3624&sid=a5a6776f735f898130eb9499adc24a17&start=40#p6369) high-quality unused building sprites for both original factions that are lying in the Aleona's Tales CVS repo. As far as I can tell, it has everything except naval structures, the Dragon Roost, Fortress and Orc towers.

The credits have the author's name: Lital Natan. No other info is available. Might be the same as [this guy](http://typoongame.com/#resume). Anyway, I thought those could be pretty fitting for _Wyrmsun_ or mods, considering their high quality and general compliance with the _Warcraft II_ style.

I've had _Aleona's Tales_ on my radar for a quite while without actually playing it -- wanted something more like WC2 visually, which _Wyrmsun_ catered to just perfectly -- but recently I decided to check out the latest version of AT, only to find out that it was broken (v2.4.3 bundled with Wargus). I managed to figure out that this was caused by faulty and/or incorrectly arranged PNG, which I was able to [fix](https://github.com/Wargus/wargus/issues/386).

It also turns out that some graphical assets are incomplete or of subpar quality, which is a shame because the game offers more than WC2, including advanced scenario scripting -- I fondly remember playing around a lot with _StarCraft_ Campaign Editor so this speaks to me (although it is my understanding that there is no proper campaign / scenario editor for Wargus / AT, everything still needs to be done by text-editing scenario files). There's generally a charming style to the game, with some fairly high quality pre-rendered and hand-drawn sprites.

On a slightly different note, I've been unable to run Kyran's _Timeless Tales_ branch from 2016 which uses _Wyrmsun_ resources alongside AT ones, mentioned [here](https://forums.stratagus.com/viewtopic.php?f=17&t=3658). I downloaded the most likely set of _Wyrmsun_ data & scripts, but it appears that this release also requires a special version of stratagus.exe that supports the scripting, which I don't know where to get, if anywhere at all.

**Comment by SenhorFlibble** [comment]
I frankensprited a new "wild" farm from _Wyrmsun_ assets: [link](https://i.imgur.com/Crp4XKx.png)

Which license can this be assigned? Should this follow the same license as the original assets I used?

**Comment by Andrettin** [comment]
@SenhorFlibble yeah, I've had great fun with Aleona's Tales in the past :) And sorry for the long delay in answering your comment!

Timeless Tales is IIRC not much different from AT, other than being rebranded. AFAIK Timeless Tales does not require Wyrmsun at all, the Wyrmsun graphics it uses are all in its repo. Just a stratagus executable should be enough (maybe the latest one would work?).

**Comment by SenhorFlibble** [comment]
> the scripts are dependent on a _Wyrmsun_ folder being inside the TT folder, as explained [here](https://forums.stratagus.com/viewtopic.php?f=17&t=3658). It is however not clear which version of _Wyrmsun_ and / or of stratagus.exe are required, and my attempts to guess had failed so far. Predictably, any Stratagus version I used crashed with errors related to _Wyrmsun_ LUA scripting, as TT apparently tries not only to use the graphics and sounds but also call _Wyrmsun_ LUA scripts as well.

I was eventually able to run _Timeless Tales_ with Wargus 2.4.3 and _Wyrmsun_ data from version 1.9.2. However, I had to make the following changes:
- comment out all references to unused resources (silver, copper and coal) in _Wyrmsun_'s scripts\anim.lua and civilization-specific anim.lua's (dwarves and gnomes). I also had to change "lumber" to "wood" in the same files
- oddly, the entire civilizations scripts directory had to be copied from the Wyrmsun\scripts directory to Wargus\scripts

Other than that, the Wyrmsun directory must be called Wyrmsun and be located in the same parent directory as the Wargus directory, e.g.

C:\Games\Wargus\
C:\Games\Wyrmsun\

I'm attaching a zip with modified _Wyrmsun_ scripts: [tales-fix.zip.txt](https://github.com/Andrettin/Wyrmsun/files/7864735/tales-fix.zip.txt)
